Source/BasicMathFunctions/arm_dot_prod_f64.c

```diff
@@ -45,33 +45,89 @@
   @param[out]    result     output result returned here.
   @return        none
  */
-
+#if defined(ARM_MATH_NEON)
```

Comment:
It will not build on 32 bit Cortex-A. You should test for __aarch64__ in addition to the ARM_MATH_NEON

Reply:
Just to make sure, does that mean that all f64 methods using neon should test for:
#if defined(ARM_MATH_NEON) && defined(__aarch64__)
Or do you have another flag that should be used in that case?

Reply:
Indeed. It is required to be able to build on a Cortex-A aarch32.
The Neon aarch64 should be selected only when the user is asking for Neon (ARM_MATH_NEON) and aarch64 is supported.
If the test is false then a pure C version should be available.

Reply:
Done! I hope I succeeded to catch them all...

Reply:
I'll try to build for aarch32. So if some are missing we will quickly see it.